src/goodbase/goodbase_cli.py

```diff
@@ -394,6 +398,14 @@ def main(
     evg_config_file = os.path.expanduser(evg_config_file)
     evg_api = RetryingEvergreenApi.get_api(config_file=evg_config_file)
 
+    if branch and git_operation == GitAction.NONE:
+        git_operation = GitAction.CHECKOUT
+        LOGGER.debug(
+            "Branch name received and no git-operation specified, setting git-operation to checkout",
+            git_operation=git_operation,
+            branch=branch,
+        )
+
```

Comment:
Can you add an additional check here that, if `git-operation` is set to anything other than `checkout` and `branch` is given, to give an error?

We don't do anything with the branch name as far as I can tell unless the operation is checkout. And giving a branch name when the operation is to merge or rebase on a certain revision doesn't seem useful to me.